### PR TITLE
Refactor task widget

### PR DESCRIPTION
**Pull request by Kasper24**
- Converted the global show signal into proper methods, show/toggle/hide
- No enable function, calling bling.widget.task_widget will return an object with the above methods
- Added the abillity to position the widget differently each time that it is showed. This with the change listed below allows it to align to the currently hovered task widget instead of having a static positon.
- Added the option to provide a wibox, widget and offset arguements to the show method, allowing to the align the task preview to another widget (possibly the hovered task widget). 
This is done by the following code:
```LUA
task_preview:show(client, {wibox = awful.screen.focused().wibar, widget = task_widget, offset = { y = 100}})
```

Minor changes:
- Removed the _fn from the placement arguement to keep it consistent with other awesome wiboxes
- Removed the 'widget' prefix from the arguement provided to the object.

Those 2 aren't that important and can be undone though.

### CODE REVIEW

**Comment by nawuko** [comment]
User Feedback, easy to configure, works like a charm with custom widgets (i have wrapped the tasklist multiple times in containers, position can still be accuratly found, awesome!) :+1: 

https://user-images.githubusercontent.com/2059141/144948685-cc31a172-5526-4885-a0fc-e34aa6ef55d7.mp4

**Comment by JavaCafe01** [comment]
@Kasper24 Looks good! I still have no Linux machine (laptop will come in a few days), so once I get it I'll test your PR and give it the thumbs up. Or, we can have @Nooo37 test it as well and merge.

**Comment by JavaCafe01** [comment]
@Kasper24 Works fine for me, just needs docs.

File: widget/task_preview.lua [hunk 1]
```diff
@@ -119,81 +152,108 @@ local function draw_widget(
...
-            task_preview_box.widget = nil
-            collectgarbage("collect")
-        end
+    return ret
 
-        if not placement_fn then
-            task_preview_box.x = s.geometry.x + widget_x
-            task_preview_box.y = s.geometry.y + widget_y
-        end
+    -- awesome.connect_signal("bling::task_preview::visibility", function(s, v, c)
```
Comment (JavaCafe01) [inline widget/task_preview.lua:227]: Might want to keep this for a few months so we can notify the user that this signal is deprecated.

**Review by JavaCafe01 — CHANGES_REQUESTED** [review]

**Comment by JavaCafe01** [comment]
Merged into #99

**Pull request closed without merging**